lib/controller.php

```diff
@@ -179,17 +181,28 @@ public function import_blob($blob) {
 			$body = wpmarkdown_markdown_to_html( $body );
 		}
 
+		$post = new WordPress_GitHub_Sync_Post( $args['ID'] );
+
 		// Can we really just mash everything together here?
+		$post_type = $post->get_type_from_path($path);
+		$post_name = $post->get_name_from_path($path);
```

Comment:
I don't think we need this method; we export the `post_type` as `layout` in the YAML frontmatter (which answers my earlier question in my code comment about mashing everything together–the answer is definitely no, as `wp_insert_post` doesn't handle the post_meta). Instead of doing this, I think we need to change the `layout` array index to `post_type`. Thoughts?